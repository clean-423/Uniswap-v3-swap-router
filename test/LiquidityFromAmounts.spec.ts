import { ethers } from 'hardhat'
import { LiquidityFromAmountsTest } from '../typechain/LiquidityFromAmountsTest'

import snapshotGasCost from './shared/snapshotGasCost'
import { encodePriceSqrt } from './shared/utilities'

describe('LiquidityFromAmounts', async () => {
  let liquidityFromAmounts: LiquidityFromAmountsTest

  before('deploy test library', async () => {
    const liquidityFromAmountsTestFactory = await ethers.getContractFactory('LiquidityFromAmountsTest')
    liquidityFromAmounts = (await liquidityFromAmountsTestFactory.deploy()) as LiquidityFromAmountsTest
  })

  describe('#getLiquidityForAmount0', () => {
    it('gas', async () => {
      const sqrtPriceAX96 = encodePriceSqrt(100, 110)
      const sqrtPriceBX96 = encodePriceSqrt(110, 100)
      await snapshotGasCost(liquidityFromAmounts.getGasCostOfGetLiquidityForAmount0(sqrtPriceAX96, sqrtPriceBX96, 100))
    })
  })

  describe('#getLiquidityForAmount1', () => {
    it('gas', async () => {
      const sqrtPriceAX96 = encodePriceSqrt(100, 110)
      const sqrtPriceBX96 = encodePriceSqrt(110, 100)
      await snapshotGasCost(liquidityFromAmounts.getGasCostOfGetLiquidityForAmount1(sqrtPriceAX96, sqrtPriceBX96, 100))
    })
  })

  describe('#getLiquidityForAmounts', () => {
    it('gas', async () => {
      const sqrtPriceAX96 = encodePriceSqrt(100, 110)
      const sqrtPriceBX96 = encodePriceSqrt(110, 100)
      await snapshotGasCost(
        liquidityFromAmounts.getGasCostOfGetLiquidityForAmounts(sqrtPriceAX96, sqrtPriceBX96, 100, 200)
      )
    })
  })
})
