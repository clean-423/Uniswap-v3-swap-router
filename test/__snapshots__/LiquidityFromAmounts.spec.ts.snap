// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiquidityFromAmounts #getLiquidityForAmount0 gas 1`] = `755`;

exports[`LiquidityFromAmounts #getLiquidityForAmount1 gas 1`] = `419`;

exports[`LiquidityFromAmounts #getLiquidityForAmounts gas 1`] = `1303`;
